### PR TITLE
Spaces delegate

### PR DESCRIPTION
**Pull request by TheOneRing**
image
image

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13062/6/1

<details>


</details>

**Comment by TheOneRing** [comment]
image

**Review by michaelstingl — APPROVED** [review]
LGTM 👍

File: src/gui/spaces/spacesmodel.cpp [hunk 1]
```diff
@@ -15,13 +15,16 @@
...
+#include "gui/guiutility.h"
 #include "gui/models/models.h"
 #include "networkjobs.h"
 
+#include <QIcon>
 #include <QPixmap>
 
 namespace {
 constexpr QSize ImageSizeC(128, 128);
+constexpr QSize ImageMarginC(ImageSizeC * 0.1);
```
Comment (fmoc) [inline src/gui/spaces/spacesmodel.cpp:27]: Even though the image size is static, I'd rather not use a percentage. This is a float calculation that needs to be rounded. I'd rather use a static offset.

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/gui/spaces/spacesbrowser.ui [hunk 2]
```diff
@@ -28,15 +28,30 @@
...
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
+     <property name="showGrid">
+      <bool>false</bool>
+     </property>
+     <property name="sortingEnabled">
+      <bool>true</bool>
+     </property>
+     <property name="wordWrap">
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/gui/spaces/spacesbrowser.ui:46]: Its set to false

File: src/gui/spaces/spacesmodel.cpp [hunk 3]
```diff
@@ -83,14 +88,19 @@ QVariant SpacesModel::data(const QModelIndex &index, int role) const
...
             return GraphApi::Drives::getDriveDisplayName(item);
-        case Columns::Description:
+        case Columns::Subtitle:
             return item.getDescription();
         case Columns::WebUrl:
             return item.getWebUrl();
         case Columns::WebDavUrl:
             return item.getRoot().getWebDavUrl();
+        case Columns::Image:
+            return {};
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/gui/spaces/spacesmodel.cpp:103]: I have to else the compiler will complain about an unnhandled case.
If we need an Image role we should probably invent one (no alias) similar to the underlying data role.

File: src/gui/spaces/spacesbrowser.ui (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/gui/spaces/spacesbrowser.ui:46]: And it has an influence on the size hint

**Review by fmoc — APPROVED** [review]

**Pull request merged**